mesheryctl/internal/cli/root/system/model.go

```diff
@@ -0,0 +1,357 @@
+
+		resp, err := utils.MakeRequest(req)
+		if err != nil {
+			utils.Log.Error(ErrConnectingToServer(err))
+			return err
+		}
+
+		// defers the closing of the response body after its use, ensuring that the resources are properly released.
+		defer resp.Body.Close()
+
+		data, err := io.ReadAll(resp.Body)
+		if err != nil {
+			utils.Log.Error(ErrInvalidAPIResponse(err))
+			return err
+		}
+
+		modelsResponse := &models.MeshmodelsAPIResponse{}
+		err = json.Unmarshal(data, modelsResponse)
+		if err != nil {
+			utils.Log.Error(ErrUnmarshallingAPIData(err))
+			return err
+		}
+
+		header := []string{"Model Name"}
+		rows := [][]string{}
+
+		for _, model := range modelsResponse.Models {
+			if len(model.DisplayName) > 0 {
+				// in `mesheryctl system model view [model-name]` command `model.Name` must be
+				// passed in order to fetch its details
+				rows = append(rows, []string{model.Name})
+			}
+		}
+
+		if len(rows) == 0 {
+			// if no model is found
+			utils.Log.Info("No model(s) found")
+		} else {
+			utils.PrintToTable(header, rows)
+		}
+
+		return nil
+	},
+}
+
+// represents the `mesheryctl system model view [model-name]` subcommand.
+var viewModelCmd = &cobra.Command{
+	Use:   "view",
+	Short: "view model",
+	Long:  "view a model queried by its name",
+	Example: `
+// View current provider
+mesheryctl system model view [model-name]
+	`,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		//Check prerequisite
+		hcOptions := &HealthCheckOptions{
+			IsPreRunE:  true,
+			PrintLogs:  false,
+			Subcommand: cmd.Use,
+		}
+		hc, err := NewHealthChecker(hcOptions)
+		if err != nil {
+			return ErrHealthCheckFailed(err)
+		}
+		// execute healthchecks
+		err = hc.RunPreflightHealthChecks()
+		if err != nil {
+			cmd.SilenceUsage = true
+			return err
+		}
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return err
+		}
+		ctx, err := mctlCfg.GetCurrentContext()
+		if err != nil {
+			return err
+		}
+		err = ctx.ValidateVersion()
+		if err != nil {
+			return err
+		}
+		return nil
+	},
+	Args: func(_ *cobra.Command, args []string) error {
+		const errMsg = "Usage: mesheryctl system model view [model-name]\nRun 'mesheryctl system model list' to see list of models"
+		if len(args) == 0 {
+			return fmt.Errorf("model name isn't specified\n\n%v", errMsg)
+		} else if len(args) > 1 {
+			return fmt.Errorf("too many arguments\n\n%v", errMsg)
+		}
+		return nil
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			log.Fatalln(err, "error processing config")
+		}
+
+		baseUrl := mctlCfg.GetBaseMesheryURL()
+		model := args[0]
+
+		url := fmt.Sprintf("%s/api/meshmodels/models/%s?pagesize=all", baseUrl, model)
+		req, err := utils.NewRequest(http.MethodGet, url, nil)
+		if err != nil {
+			utils.Log.Error(ErrGettingRequestContext(err))
+			return err
+		}
+
+		client := &http.Client{}
+		resp, err := client.Do(req)
```

Comment:
```suggestion
		resp, err := utils.MakeRequest(req)
```
Make use of this function

Reply:
@suhail34  ok